app/models/krikri/validation_report.rb

```diff
@@ -0,0 +1,54 @@
+module Krikri
+  class ValidationReport
+    attr_accessor :provider_id, :start, :rows
+
+    REQUIRED_FIELDS = ['dataProvider_name', 'isShownAt_id', 'preview_id',
+                       'sourceResource_rights', 'sourceResource_title',
+                       'sourceResource_type_id']
+
+    ##
+    # @param &block may contain provider_id
+    #   Sample use:
+    #     ValidationReport.new.all do
+    #       self.provider_id = '0123'
+    #     end
+    #
+    # @return [Array<Blacklight::SolrResponse::Facet>]
+    def all(&block)
+      # set values from block
+      instance_eval &block if block_given?
+
+      query_params = { :rows => 0,
+                       'facet.field' => REQUIRED_FIELDS,
+                       'facet.mincount' => 10000000,
+                       'facet.missing' => true }
+      query_params[:fq] = "provider_id:\"#{@provider_id}\"" if
+        @provider_id.present?
```

Comment:
The source of my confusion is this:  When I load `http://localhost:3000/krikri/reports?provider=123` in my browser window, under "Validation Reports", I see "sourceResource_title missing (1 record)".  That is not correct.  It should say "sourceResource_title missing (0 records)".